eZ/Publish/Core/MVC/Symfony/Controller/Content/DownloadController.php

```diff
@@ -50,6 +52,10 @@ public function downloadBinaryFileAction($contentId, $fieldIdentifier, $filename
             $content = $this->contentService->loadContent($contentId);
         }
 
+        if ($content->contentInfo->isTrashed()) {
+            throw new NotFoundException('File', $filename);
+        }
+
```

Comment:
So, if user has access to the trashed content, should he not be able to download the file?

Reply:
@andrerom The only way he could download this file (in case he has access to trash and download) is by going directly to the file URL. He won't be able to access it neither on frontend nor on BackOffice as the download link doesn't appear in those places after Content is trashed. So he would have to know the URL before the Content is deleted, save it somewhere and go directly to it after trashing Content - I'm not sure anyone would do this.